diff -r 53d1f20e8169 configure.ac
--- a/configure.ac	Wed Jun 19 17:25:56 2013 +0200
+++ b/configure.ac	Tue Jun 25 22:20:56 2013 +0200
@@ -2753,6 +2753,15 @@
 and contains `psp_iticksperclktick'.])],,
                 [#include <sys/pstat.h>])
 
+# Reasons for checking:
+#   lconv decimal_point - not in android
+#
+AC_CHECK_MEMBER(struct lconv.decimal_point,
+                [AC_DEFINE(HAVE_LCONV_DECIMAL_POINT, 1,
+[Define to 1 if <locale.h> `struct lconv' exists
+and contains `decimal_point'.])],,
+                [#include <locale.h>])
+
 # C++ tests, when required
 #
 if test $enable_cxx = yes; then
diff -r 53d1f20e8169 gmp-impl.h
--- a/gmp-impl.h	Wed Jun 19 17:25:56 2013 +0200
+++ b/gmp-impl.h	Tue Jun 25 22:20:56 2013 +0200
@@ -4368,7 +4368,8 @@
 #define GMP_DECIMAL_POINT  (nl_langinfo (RADIXCHAR))
 #endif
 /* localeconv is slower since it returns all locale stuff */
-#if HAVE_LOCALECONV && ! defined (GMP_DECIMAL_POINT)
+/* A placeholder localeconv is useless. */
+#if HAVE_LOCALECONV && HAVE_LCONV_DECIMAL_POINT && ! defined (GMP_DECIMAL_POINT)
 #define GMP_DECIMAL_POINT  (localeconv()->decimal_point)
 #endif
 #if ! defined (GMP_DECIMAL_POINT)
