--- a/pthread_cancel.c
+++ b/pthread_cancel.c
@@ -67,12 +67,12 @@ ptw32_cancel_callback (ULONG_PTR unused)
 DWORD
 ptw32_Registercancellation (PAPCFUNC unused1, HANDLE threadH, DWORD unused2)
 {
-  CONTEXT context;
+  /*CONTEXT context;
 
   context.ContextFlags = CONTEXT_CONTROL;
   GetThreadContext (threadH, &context);
   PTW32_PROGCTR (context) = (DWORD_PTR) ptw32_cancel_self;
-  SetThreadContext (threadH, &context);
+  SetThreadContext (threadH, &context);*/
   return 0;
 }
 
--- a/pthread_win32_attach_detach_np.c
+++ b/pthread_win32_attach_detach_np.c
@@ -89,11 +89,11 @@ pthread_win32_process_attach_np ()
   }
 #else
 #  if ! defined(WINCE)
-  if(GetSystemDirectory(QuserExDLLPathBuf, sizeof(QuserExDLLPathBuf)/sizeof(TCHAR)) &&
+  /*if(GetSystemDirectory(QuserExDLLPathBuf, sizeof(QuserExDLLPathBuf)/sizeof(TCHAR)) &&
       0 == _tcsncat_s(QuserExDLLPathBuf, _countof(QuserExDLLPathBuf), TEXT("\\QUSEREX.DLL"), 12))
     {
       ptw32_h_quserex = LoadLibrary(QuserExDLLPathBuf);
-    }
+    }*/
 #  endif
 #endif
 
-- 
2.10.2.windows.1

