From 1b0fbedb16f7fa00be3e7da06e541592b0ce4819 Mon Sep 17 00:00:00 2001
From: David Sugar <dyfet@gnutelephony.org>
Date: Sat, 13 Sep 2014 05:43:35 -0700
Subject: [PATCH] any addr and overloads

---
 commoncpp/socket.cpp | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/commoncpp/socket.cpp b/commoncpp/socket.cpp
index 0834aa8..9dcd975 100644
--- a/commoncpp/socket.cpp
+++ b/commoncpp/socket.cpp
@@ -868,10 +868,13 @@ ucommon::Socket::address Socket::getSender() const
 IPV4Host Socket::getIPV4Sender(tpport_t *port) const
 {
     sockaddr_in* from = getSender();
+    struct in_addr any;
+    any.s_addr = INADDR_ANY;
+
     if (from == NULL) {
         if (port)
             port = 0;
-        return IPV4Host(INADDR_ANY);
+        return IPV4Host(any);
     }
 
     if (port)
@@ -912,10 +915,13 @@ ucommon::Socket::address Socket::getLocal() const
 IPV4Host Socket::getIPV4Local(tpport_t *port) const
 {
     sockaddr_in* from = getLocal();
+    struct in_addr any;
+    any.s_addr = INADDR_ANY;
+
     if (from == NULL) {
         if (port)
             port = 0;
-        return IPV4Host(INADDR_ANY);
+        return IPV4Host(any);
     }
 
     if (port)
@@ -962,10 +968,13 @@ ucommon::Socket::address Socket::getPeer() const
 IPV4Host Socket::getIPV4Peer(tpport_t *port) const
 {
     sockaddr_in* from = getPeer();
+    struct in_addr any;
+    any.s_addr = INADDR_ANY;
+
     if (from == NULL) {
         if (port)
             port = 0;
-        return IPV4Host(INADDR_ANY);
+        return IPV4Host(any);
     }
 
     if (port)
-- 
1.9.3

