From 9ccc6e7acb2ec0805c319710d370747aa275bb02 Mon Sep 17 00:00:00 2001
From: Socapex <philippe_groarke@yahoo.ca>
Date: Thu, 9 Oct 2014 16:30:49 -0400
Subject: [PATCH] OSX: Didn't find headers if already installed.

---
 gnutls/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnutls/Makefile.am b/gnutls/Makefile.am
index b11fa08..439af7e 100644
--- a/gnutls/Makefile.am
+++ b/gnutls/Makefile.am
@@ -10,7 +10,7 @@
 
 MAINTAINERCLEANFILES = Makefile.in Makefile
 RELEASE = -version-info $(LT_VERSION)
-AM_CXXFLAGS = -I$(top_srcdir)/inc @UCOMMON_FLAGS@
+AM_CXXFLAGS = -I$(top_srcdir)/inc @UCOMMON_FLAGS@ @GNUTLS_CFLAGS@
 
 noinst_HEADERS = local.h
 
