 src/transports/smart_protocol.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/transports/smart_protocol.c b/src/transports/smart_protocol.c
index c01656dc4..9a6684bfd 100644
--- a/src/transports/smart_protocol.c
+++ b/src/transports/smart_protocol.c
@@ -229,6 +229,7 @@ static int recv_pkt(git_pkt **out_pkt, git_pkt_type *out_type, gitno_buffer *buf
 			return error;
 
 		if ((ret = gitno_recv(buf)) < 0) {
+			printf("@@@ pkt: %s - error %u", pkt, ret);
 			return ret;
 		} else if (ret == 0) {
 			git_error_set(GIT_ERROR_NET, "early EOF");
@@ -336,7 +337,7 @@ int git_smart__negotiate_fetch(git_transport *transport, git_repository *repo, c
 		if (error < 0) {
 			if (GIT_ITEROVER == error)
 				break;
-
+			printf("@@@ Error when parsing: %s %u", git_oid_tostr_s(&oid), error);
 			goto on_error;
 		}
 
