From 3d991830d5c566d84ec14da9e681a51f1a72a04d Mon Sep 17 00:00:00 2001
From: jrun <darwinskernel@gmail.com>
Date: Thu, 27 Feb 2020 14:10:54 -0500
Subject: [PATCH 1/4] win_config

---
 pjlib/include/pj/config_site.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pjlib/include/pj/config_site.h b/pjlib/include/pj/config_site.h
index 8c54e4e09..e99ecc95b 100644
--- a/pjlib/include/pj/config_site.h
+++ b/pjlib/include/pj/config_site.h
@@ -20,3 +20,7 @@
 #define PJ_ICE_COMP_BITS                        2
 #define PJ_ICE_MAX_CHECKS                       1024
 
+/*
+ * WINDOWS settings.
+ */
+#define THIRD_PARTY_MEDIA			0
-- 
2.24.1

