From e6f1d40bbb7e4aeb1892939b5cf297df12a3b8e2 Mon Sep 17 00:00:00 2001
From: jrun <darwinskernel@gmail.com>
Date: Wed, 4 Mar 2020 14:03:16 -0500
Subject: [PATCH 4/4] win include winsock2 for tcpkeeps

---
 pjlib/src/pj/sock_bsd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pjlib/src/pj/sock_bsd.c b/pjlib/src/pj/sock_bsd.c
index 33a4f21c3..58b78cf1d 100644
--- a/pjlib/src/pj/sock_bsd.c
+++ b/pjlib/src/pj/sock_bsd.c
@@ -35,6 +35,8 @@
 # if !defined(TCP_KEEPIDLE) && defined(TCP_KEEPALIVE)
 #  define TCP_KEEPIDLE TCP_KEEPALIVE
 # endif
+#else
+#include <Winsock2.h>
 #endif
 
 /*
-- 
2.24.1

