From 42035112f3101940164d04f37cd125dac59d13aa Mon Sep 17 00:00:00 2001
From: jrun <darwinskernel@gmail.com>
Date: Thu, 20 Feb 2020 13:21:15 -0500
Subject: [PATCH 09/12] sip_config

---
 pjlib/include/pj/config_site.h | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/pjlib/include/pj/config_site.h b/pjlib/include/pj/config_site.h
index d7b7ff82c..f51eef004 100644
--- a/pjlib/include/pj/config_site.h
+++ b/pjlib/include/pj/config_site.h
@@ -5,7 +5,13 @@
 /*
 * PJNAT settings.
 */
-#define PJ_ICE_MAX_CAND     256
-#define PJ_ICE_MAX_STUN     6
-#define PJ_ICE_COMP_BITS    2
-#define PJ_ICE_MAX_CHECKS   1024
+#define PJ_ICE_MAX_CAND                    256
+#define PJ_ICE_MAX_STUN                    6
+#define PJ_ICE_COMP_BITS                   2
+#define PJ_ICE_MAX_CHECKS                  1024
+
+/*
+* PJSIP settings.
+*/
+#define PJSIP_MAX_PKT_LEN                  8000
+#define PJSIP_TRANSPORT_SERVER_IDLE_TIME   3
-- 
2.24.1

