From c9617dac4059d5346977a8d18699d6a9693c7e7c Mon Sep 17 00:00:00 2001
From: jrun <darwinskernel@gmail.com>
Date: Fri, 21 Feb 2020 11:55:37 -0500
Subject: [PATCH 11/12] fix_turn_fallback

---
 pjnath/src/pjnath/turn_session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pjnath/src/pjnath/turn_session.c b/pjnath/src/pjnath/turn_session.c
index 2da1a3651..99877a72b 100644
--- a/pjnath/src/pjnath/turn_session.c
+++ b/pjnath/src/pjnath/turn_session.c
@@ -659,7 +659,7 @@ PJ_DEF(pj_status_t) pj_turn_session_set_server( pj_turn_session *sess,
 	PJ_ASSERT_RETURN(default_port>0 && default_port<65536, PJ_EINVAL);
 	sess->default_port = (pj_uint16_t)default_port;
 
-	cnt = PJ_TURN_MAX_DNS_SRV_CNT;
+	cnt = 1;
 	ai = (pj_addrinfo*)
 	     pj_pool_calloc(sess->pool, cnt, sizeof(pj_addrinfo));
 
-- 
2.24.1

