From 328a6837cc739440ed9c9abe18a43a36d3c4a60d Mon Sep 17 00:00:00 2001
From: sauwming <ming@teluu.com>
Date: Mon, 30 Aug 2021 11:45:52 +0800
Subject: [PATCH 1/2] Resort ICE checklist upon entering nomination stage
 (#2806)

---
 pjnath/src/pjnath/ice_session.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pjnath/src/pjnath/ice_session.c b/pjnath/src/pjnath/ice_session.c
index 3fecc3def..ef88807a1 100644
--- a/pjnath/src/pjnath/ice_session.c
+++ b/pjnath/src/pjnath/ice_session.c
@@ -3463,7 +3463,11 @@ static void handle_incoming_check(pj_ice_sess *ice,
 	LOG4((ice->obj_name, "New triggered check added: %d", 
 	     ice->clist.count));
 	pj_log_push_indent();
+
 	perform_check(ice, &ice->clist, ice->clist.count++, nominate);
+    	/* Re-sort the list because of the newly added pair. */
+    	sort_checklist(ice, &ice->clist);
+
 	pj_log_pop_indent();
 
     } else {
-- 
2.25.1

