 pjnath/src/pjnath/ice_session.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pjnath/src/pjnath/ice_session.c b/pjnath/src/pjnath/ice_session.c
index d847c3c0c..9b668f45b 100644
--- a/pjnath/src/pjnath/ice_session.c
+++ b/pjnath/src/pjnath/ice_session.c
@@ -4138,7 +4138,11 @@ static void handle_incoming_check(pj_ice_sess *ice,
 	LOG4((ice->obj_name, "New triggered check added: %d", 
 	     ice->clist.count));
 	pj_log_push_indent();
+
 	perform_check(ice, &ice->clist, ice->clist.count++, nominate);
+	/* Re-sort the list because of the newly added pair. */
+	sort_checklist(ice, &ice->clist);
+
 	pj_log_pop_indent();
 
     } else {
-- 
2.25.1

