diff --git a/pjlib/src/pj/ip_helper_generic.c b/pjlib/src/pj/ip_helper_generic.c
index ab441a504..ca569d374 100644
--- a/pjlib/src/pj/ip_helper_generic.c
+++ b/pjlib/src/pj/ip_helper_generic.c
@@ -59,6 +59,15 @@
 #else
 #   define TRACE_(exp)
 #endif
+static const char *get_addr(void *addr)
+{
+static char txt[PJ_INET6_ADDRSTRLEN];
+struct sockaddr *ad = (struct sockaddr*)addr;
+if (ad->sa_family != PJ_AF_INET && ad->sa_family != PJ_AF_INET6)
+	return "?";
+return pj_inet_ntop2(ad->sa_family, pj_sockaddr_get_addr(ad), 
+				txt, sizeof(txt));
+}
 
 
 #if 0
@@ -124,8 +133,10 @@ static pj_status_t if_enum_by_af(int af,
 	/* Ignore 192.0.0.0/29 address.
 	 * Ref: https://datatracker.ietf.org/doc/html/rfc7335#section-4
 	 */
+	uint32_t ntohl = pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4;
+	printf("@@@ %s %u\n", get_addr(ad), ntohl);
 	if (af==pj_AF_INET() &&
-	    (pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4) ==
+	    ntohl ==
 	     201326592) /* 0b1100000000000000000000000000 which is
 	                   192.0.0.0 >> 4 */
 	{
@@ -241,6 +252,8 @@ static pj_status_t if_enum_by_af(int af,
 	/* Ignore 192.0.0.0/29 address.
 	 * Ref: https://datatracker.ietf.org/doc/html/rfc7335#section-4
 	 */
+	uint32_t ntohl = pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4;
+	printf("@@@2 %s %u\n", get_addr(ad), ntohl);
 	if (af==pj_AF_INET() &&
 	    (pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4) ==
 	     201326592) /* 0b1100000000000000000000000000 which is
@@ -353,6 +366,8 @@ static pj_status_t if_enum_by_af(int af, unsigned *p_cnt, pj_sockaddr ifs[])
 	/* Ignore 192.0.0.0/29 address.
 	 * Ref: https://datatracker.ietf.org/doc/html/rfc7335#section-4
 	 */
+	uint32_t ntohl = pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4;
+	printf("@@@3 %s %u\n", get_addr(ad), ntohl);
 	if (af==pj_AF_INET() &&
 	    (pj_ntohl(((pj_sockaddr_in*)ad)->sin_addr.s_addr) >> 4) ==
 	     201326592) /* 0b1100000000000000000000000000 which is
