 pjnath/src/pjnath/ice_strans.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pjnath/src/pjnath/ice_strans.c b/pjnath/src/pjnath/ice_strans.c
index 7583bbf8..57cf9699 100644
--- a/pjnath/src/pjnath/ice_strans.c
+++ b/pjnath/src/pjnath/ice_strans.c
@@ -1918,8 +1918,9 @@ static pj_status_t ice_close_tcp_connection(pj_ice_sess *ice,
 	pj_ice_strans *ice_st = (pj_ice_strans *)ice->user_data;
 	pj_ice_strans_comp *st_comp = ice_st->comp[lcand->comp_id - 1];
 	if (st_comp->stun[0].sock) {
-		return pj_stun_sock_close(st_comp->stun[0].sock, check_id);
-	}
+          const pj_ice_sess_cand *rcand = check->rcand;
+          return pj_stun_sock_close(st_comp->stun[0].sock, &rcand->addr);
+	} 
 
 	return PJ_EINVAL;
 }
