 pjnath/src/pjnath/ice_session.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pjnath/src/pjnath/ice_session.c b/pjnath/src/pjnath/ice_session.c
index 12b8dad06..dea1cb581 100644
--- a/pjnath/src/pjnath/ice_session.c
+++ b/pjnath/src/pjnath/ice_session.c
@@ -30,13 +30,17 @@
 #include <pj/rand.h>
 #include <pj/string.h>
 
-#if defined(_WIN32) || defined(__APPLE__)
+#if defined(_WIN32)
 /* TODO(sblin): find an alternative for these paltforms */
 #else
 /* The following headers are used to get DEPRECATED addresses
  * as specified in RFC 2462 Section 5.5.4
  * https://tools.ietf.org/html/rfc2462#section-5.5.4
  */
+#if defined(__APPLE__)
+#include <net/if.h>
+#include <netinet6/in6_var.h>
+#else
 #include <arpa/inet.h>
 #include <asm/types.h>
 #include <linux/netlink.h>
@@ -44,6 +48,7 @@
 #include <sys/socket.h>
 #include <unistd.h>
 #endif
+#endif
 
 /* String names for candidate types */
 static const char *cand_type_names[] =
