--- a/build/vs/pjproject-vs14-common-config.props
+++ b/build/vs/pjproject-vs14-common-config.props
@@ -18,22 +18,22 @@
   <PropertyGroup>    
     <API_Family Condition="'$(API_Family)'==''">WinDesktop</API_Family>
     <PreprocessorDef></PreprocessorDef>
-    <DefaultToolset>v140</DefaultToolset>
+    <DefaultToolset>v141</DefaultToolset>
   </PropertyGroup>  
   <Choose>
     <When Condition="'$(API_Family)'=='WinDesktop'">
       <PropertyGroup>
-        <BuildToolset>v140</BuildToolset>
+        <BuildToolset>v141</BuildToolset>
         <PreprocessorDef Condition="'$(Platform)'=='Win32'">WIN32;PJ_WIN32=1;PJ_M_I386=1;</PreprocessorDef>
         <PreprocessorDef Condition="'$(Platform)'=='x64'">WIN64;PJ_WIN64=1;PJ_M_X86_64=1;</PreprocessorDef>
       </PropertyGroup>
     </When>
     <When Condition="'$(API_Family)'=='UWP'">
       <PropertyGroup>
-        <BuildToolset>v140</BuildToolset>
+        <BuildToolset>v141</BuildToolset>
         <PreprocessorDef>PJ_WIN32_UWP;UNICODE;_UNICODE;</PreprocessorDef>
         <PreprocessorDef Condition="'$(Platform)'=='ARM'">$(PreprocessorDef);PJ_M_ARMV7=1;</PreprocessorDef>
-        <PlatformVersion>10.0.10586.0</PlatformVersion>
+        <PlatformVersion>10.0.16299.0</PlatformVersion>
         <MinPlatformVersion>10.0.10240.0</MinPlatformVersion>
         <AppTypeRev>10.0</AppTypeRev>
         
--- a/build/vs/pjproject-vs14-common-defaults.props
+++ b/build/vs/pjproject-vs14-common-defaults.props
@@ -3,7 +3,7 @@
   <ImportGroup Label="PropertySheets">
   </ImportGroup>
   <PropertyGroup Label="UserMacros">
-    <VSVer>14</VSVer>
+    <VSVer>15</VSVer>
   </PropertyGroup>
   <PropertyGroup>
     <_ProjectFileVersion>14.0.22823.1</_ProjectFileVersion>
-- 
2.10.2.windows.1

