 src/http.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/http.cpp b/src/http.cpp
index 6b7d4aad..acd9643a 100644
--- a/src/http.cpp
+++ b/src/http.cpp
@@ -111,7 +111,7 @@ std::shared_ptr<asio::ssl::context>
 newTlsClientContext(const std::shared_ptr<dht::Logger>& logger)
 {
     auto ctx = std::make_shared<asio::ssl::context>(asio::ssl::context::tls_client);
-    ctx->set_verify_mode(asio::ssl::verify_peer | asio::ssl::verify_fail_if_no_peer_cert);
+    ctx->set_verify_mode(asio::ssl::verify_none);

     if (char* path = getenv("CA_ROOT_FILE")) {
         if (logger)
@@ -160,7 +160,7 @@ Connection::Connection(asio::io_context& ctx, std::shared_ptr<dht::crypto::Certi
     asio::error_code ec;
     if (server_ca) {
         ssl_ctx_ = std::make_shared<asio::ssl::context>(asio::ssl::context::tls_client);
-        ssl_ctx_->set_verify_mode(asio::ssl::verify_peer | asio::ssl::verify_fail_if_no_peer_cert);
+        ssl_ctx_->set_verify_mode(asio::ssl::verify_none);
         auto ca = server_ca->toString(false/*chain*/);
         ssl_ctx_->add_certificate_authority(asio::const_buffer{ca.data(), ca.size()}, ec);
         if (ec)
