 include/opendht/default_types.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/opendht/default_types.h b/include/opendht/default_types.h
index ef3434c..9b005a2 100644
--- a/include/opendht/default_types.h
+++ b/include/opendht/default_types.h
@@ -144,17 +144,18 @@ public:
     static const ValueType TYPE;
 
     TrustRequest() {}
-    TrustRequest(std::string s) : service(s) {}
-    TrustRequest(std::string s, const Blob& d) : service(s), payload(d) {}
+    TrustRequest(std::string s, std::string ci) : service(s), conversationId(ci) {}
+    TrustRequest(std::string s, std::string ci, const Blob& d) : service(s), conversationId(ci), payload(d) {}
 
     static Value::Filter getFilter() {
         return EncryptedValue::getFilter();
     }
 
     std::string service;
+    std::string conversationId;
     Blob payload;
     bool confirm {false};
-    MSGPACK_DEFINE_MAP(service, payload, confirm)
+    MSGPACK_DEFINE_MAP(service, conversationId, payload, confirm)
 };
 
 class OPENDHT_PUBLIC IceCandidates : public EncryptedValue<IceCandidates>
