--- yaml-cpp/CMakeLists.txt.orig	2014-09-05 18:09:22.644410515 -0400
+++ yaml-cpp/CMakeLists.txt	2014-09-05 18:12:20.372407142 -0400
@@ -269,10 +269,10 @@
	FILES_MATCHING PATTERN "*.h"
 )

-if(UNIX)
+if(UNIX OR MINGW)
	set(PC_FILE ${CMAKE_BINARY_DIR}/yaml-cpp.pc)
	configure_file("yaml-cpp.pc.cmake" ${PC_FILE} @ONLY)
-	install(FILES ${PC_FILE} DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
+	install(FILES ${PC_FILE} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
 endif()
