--- a/mpi/config.links	2013-07-25 13:10:04.000000000 +0400
+++ b/mpi/config.links	2013-11-10 15:36:20.000000000 +0400
@@ -100,6 +100,12 @@
 	cat  $srcdir/mpi/i386/syntax.h	 >>./mpi/asm-syntax.h
 	path="i586 i386"
 	;;
+    x86_64-apple-darwin13*)
+    echo '#define BSD_SYNTAX' >>./mpi/asm-syntax.h
+    cat  $srcdir/mpi/i386/syntax.h      >>./mpi/asm-syntax.h
+    path="amd64"
+        mpi_cpu_arch="x86"
+    ;;
     i[3467]86*-*-*)
 	echo '#define ELF_SYNTAX' >>./mpi/asm-syntax.h
 	cat  $srcdir/mpi/i386/syntax.h	    >>./mpi/asm-syntax.h
