From a089e8f34814b5568c96bee5695028fb5fd53ac6 Mon Sep 17 00:00:00 2001
From: kkostiuk <kateryna.kostiuk@savoirfairelinux.com>
Date: Wed, 2 Jun 2021 09:17:09 -0400
Subject: [PATCH] changes

---
 upnp/src/genlib/miniserver/miniserver.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/upnp/src/genlib/miniserver/miniserver.c b/upnp/src/genlib/miniserver/miniserver.c
index 7244802..471b8ee 100644
--- a/upnp/src/genlib/miniserver/miniserver.c
+++ b/upnp/src/genlib/miniserver/miniserver.c
@@ -431,6 +431,8 @@ static void RunMiniServer(
 
 	gMServState = MSERV_RUNNING;
 	while (!stopSock) {
+		printf("@@@@@@RunMiniServer");
+		fflush(stdout);
 		FD_ZERO(&rdSet);
 		FD_ZERO(&expSet);
 		/* FD_SET()'s */
@@ -452,6 +454,8 @@ static void RunMiniServer(
 			continue;
 		}
 		if (ret == SOCKET_ERROR) {
+		printf("@@@@@@ RunMiniServer error, %s", errorBuffer);
+		fflush(stdout);
 			strerror_r(errno, errorBuffer, ERROR_BUFFER_LEN);
 			UpnpPrintf(UPNP_CRITICAL,
 				SSDP,
@@ -639,6 +643,8 @@ static int get_miniserver_sockets(
 	listenfd6UlaGua = socket(AF_INET6, SOCK_STREAM, 0);
 	if (listenfd6UlaGua == INVALID_SOCKET) {
 		strerror_r(errno, errorBuffer, ERROR_BUFFER_LEN);
+		printf("@@@@@@ webserver error, %s", errorBuffer);
+		fflush(stdout);
 		UpnpPrintf(UPNP_INFO,
 			MSERV,
 			__FILE__,
-- 
2.31.1

