--- a/upnp/src/inet_pton.c	2013-11-15 11:10:05.000000000 -0500
+++ b/upnp/src/inet_pton.c	2016-02-16 13:21:03.963801774 -0500
@@ -304,20 +304,7 @@
 	/* NOTREACHED */
 }

-int inet_pton(int af, const char *src, void *dst)
-{
-	switch (af) {
-	case AF_INET:
-		return inet_pton4(src, dst);
-#ifdef INET_IPV6
-	case AF_INET6:
-		return inet_pton6(src, dst);
-#endif
-	default:
-		/*__set_errno(EAFNOSUPPORT);*/
-		return -1;
-	}
-	/* NOTREACHED */
-}
+#define InetPtonA inet_pton
+WINSOCK_API_LINKAGE INT WSAAPI InetPtonA(INT Family, LPCSTR pStringBuf, PVOID pAddr);

 #endif /* WIN32 */
