From 649b319cfa3fc9bb4b9a93264cff70c187336093 Mon Sep 17 00:00:00 2001
From: philippegorley <gorley.philippe@gmail.com>
Date: Fri, 28 Jul 2017 09:45:23 -0400
Subject: [PATCH] use internal log2f

---
 configure | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index ec9389d..02b05c7 100755
--- a/configure
+++ b/configure
@@ -1033,9 +1033,9 @@ if [ "$thread" = "posix" ]; then
 fi
 [ "$thread" != "no" ] && define HAVE_THREAD
 
-if cc_check "math.h" "-Werror" "return log2f(2);" ; then
-    define HAVE_LOG2F
-fi
+#if cc_check "math.h" "-Werror" "return log2f(2);" ; then
+    define HAVE_LOG2F 0
+#fi
 
 if [ "$SYS" != "WINDOWS" ] && cpp_check "sys/mman.h unistd.h" "" "defined(MAP_PRIVATE)"; then
     define HAVE_MMAP
-- 
2.7.4

