--- a/SMP/config.h
+++ b/SMP/config.h
@@ -22,7 +22,7 @@
 #define HAVE_GPL 1
 #define HAVE_INTERLACED 1
 #if BIT_DEPTH==8
-#   define HAVE_OPENCL 1
+#   define HAVE_OPENCL 0
 #else
 #   define HAVE_OPENCL 0
 #endif
--- a/SMP/config.h
+++ b/SMP/config.h
@@ -48,4 +48,6 @@
 #endif
 #define HAVE_WINRT 0
 #define HAVE_BITDEPTH8 1
-#define HAVE_BITDEPTH10 1
\ No newline at end of file
+#define HAVE_BITDEPTH10 1
+#undef HAVE_WINRT
+#define HAVE_WINRT 1
--- a/SMP/libx264.vcxproj
+++ b/SMP/libx264.vcxproj
@@ -1351,6 +1351,8 @@ del /f /q $(OutDir)\licenses\x264.txt
     <ClCompile Include="..\common\mc.c" />
     <ClCompile Include="..\common\mvpred.c" />
     <ClCompile Include="..\common\opencl.c">
+			<ExcludedFromBuild Condition="'$(Platform)'=='x64'">true</ExcludedFromBuild>
+			<ExcludedFromBuild Condition="'$(Platform)'=='x86'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)'=='Debug_10b'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)'=='Release_10b'">true</ExcludedFromBuild>
     </ClCompile>
\ No newline at end of file
-- 
2.10.2.windows.1

