--- a/cmake/modules/Findkashmir.cmake	2017-06-02 12:01:30.415787179 -0400
+++ b/cmake/modules/Findkashmir.cmake	2017-06-02 12:01:33.329120391 -0400
@@ -1,11 +1,2 @@
-# Copyright 2014-2016, Corvusoft Ltd, All Rights Reserved.
-
-find_path( kashmir_INCLUDE kashmir HINTS "${PROJECT_SOURCE_DIR}/dependency/kashmir" "/usr/include" "/usr/local/include" "/opt/local/include" )
-
-if ( kashmir_INCLUDE )
-    set( KASHMIR_FOUND TRUE )
-
-    message( STATUS "${Green}Found Kashmir include at: ${kashmir_INCLUDE}${Reset}" )
-else ( )
-    message( FATAL_ERROR "${Red}Failed to locate Kashmir dependency.${Reset}" )
-endif ( )
+set( KASHMIR_FOUND TRUE )
+set( kashmir_INCLUDE "${PROJECT_SOURCE_DIR}/dependency/kashmir" )
