--- a/source/corvusoft/restbed/uri.cpp	2016-04-14 21:38:26.000000000 +0000
+++ b/source/corvusoft/restbed/uri.cpp	2016-09-30 18:55:09.304594102 +0000
@@ -8,9 +8,9 @@
 #include <cstdlib>
 #include <stdexcept>
 
-#if defined(WIN32)
+#if defined(_WIN32)
     #include <ciso646>
-    #include <Winsock2.h>
+    #include <winsock2.h>
     #pragma comment( lib, "Ws2_32.lib" )
 #else
     #include <netdb.h>
