 configure.ac          | 5 +++++
 lib/system/fastopen.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e81ff8970..2c8635099 100644
--- a/configure.ac
+++ b/configure.ac
@@ -402,6 +402,11 @@ fi
 
 AM_CONDITIONAL(WANT_TEST_SUITE, test "$full_test_suite" = "yes")
 
+dnl check is connectx is present
+AC_CHECK_FUNCS([connectx],
+               [AC_DEFINE(HAVE_VALID_CONNECTX, 1, [Enable connectx])],
+               [AC_DEFINE(HAVE_VALID_CONNECTX, 0, [Disable connectx])])
+
 dnl GCC warnings to enable
 
 AC_ARG_ENABLE([gcc-warnings],
diff --git a/lib/system/fastopen.c b/lib/system/fastopen.c
index 0f673b7c4..7b4ea0109 100644
--- a/lib/system/fastopen.c
+++ b/lib/system/fastopen.c
@@ -36,7 +36,7 @@
 #endif
 
 /* TCP Fast Open on OSX behaves differently from Linux, so define these helpers */
-#if defined __APPLE__ && defined __MACH__ && defined CONNECT_DATA_IDEMPOTENT && defined CONNECT_RESUME_ON_READ_WRITE
+#if defined __APPLE__ && defined __MACH__ && defined CONNECT_DATA_IDEMPOTENT && defined CONNECT_RESUME_ON_READ_WRITE && HAVE_CONNECTX
 # define TCP_FASTOPEN_OSX
 #elif defined TCP_FASTOPEN && defined MSG_FASTOPEN
 # define TCP_FASTOPEN_LINUX

