--- gnutls-3.4.0/lib/x509/output.c	2015-05-04 12:21:29.453100915 -0400
+++ gnutls-3.4.0/lib/x509/output.c	2015-05-04 12:32:27.737226031 -0400
@@ -1140 +1140 @@
-					addf(str, _("\t\tP: "));
+					adds(str, _("\t\tP: "));
@@ -1145 +1145 @@
-					addf(str, _("\t\tQ: "));
+					adds(str, _("\t\tQ: "));
@@ -1150 +1150 @@
-					addf(str, _("\t\tG: "));
+					adds(str, _("\t\tG: "));
--- gnutls-3.4.0/lib/x509/ocsp_output.c
+++ gnutls-3.4.0/lib/x509/ocsp_output.c
@@ -291 +291 @@
-					addf(str, _("\tResponder Key ID: "));
+					adds(str, _("\tResponder Key ID: "));