diff --git a/lib/verify-tofu.c b/lib/verify-tofu.c
index 1b92583..3c0dd8d 100644
--- a/lib/verify-tofu.c
+++ b/lib/verify-tofu.c
@@ -493,7 +493,7 @@ int gnutls_store_pubkey(const char *db_name, gnutls_tdb_t tdb, const char *host,
 			return gnutls_assert_val(ret);
 
 		_gnutls_debug_log("Configuration path: %s\n", local_file);
-		mkdir(local_file, 0700);
+		mkdir(local_file);
 
 		ret = find_config_file(local_file, sizeof(local_file));
 		if (ret < 0)
@@ -594,7 +594,7 @@ int gnutls_store_commitment(const char *db_name, gnutls_tdb_t tdb,
 			return gnutls_assert_val(ret);
 
 		_gnutls_debug_log("Configuration path: %s\n", local_file);
-		mkdir(local_file, 0700);
+		mkdir(local_file);
 
 		ret = find_config_file(local_file, sizeof(local_file));
 		if (ret < 0)
