From 2c0b6f5e5ba04d7a74e8406a3df1fd433680599d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Adrien=20B=C3=A9raud?= <adrien.beraud@savoirfairelinux.com>
Date: Fri, 1 Nov 2019 10:51:08 -0400
Subject: [PATCH] look for official http-parser

---
 dev/CMakeLists.txt | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/dev/CMakeLists.txt b/dev/CMakeLists.txt
index 515e8ba7..baf8797a 100644
--- a/dev/CMakeLists.txt
+++ b/dev/CMakeLists.txt
@@ -86,11 +86,8 @@ IF (RESTINIO_MASTER_PROJECT)
 		ENDIF ()
 
 		# HTTP parser
-		IF (RESTINIO_USE_EXTERNAL_HTTP_PARSER)
-			find_package(http-parser REQUIRED)
-		ELSE ()
-			find_package(unofficial-http-parser CONFIG REQUIRED)
-		ENDIF ()
+		find_path(http-parser_INCLUDE http_parser.h)
+		find_library(http-parser_LIBRARY http_parser)
 
 		# fmtlib
 		find_package(fmt REQUIRED)
@@ -100,11 +97,7 @@ IF (RESTINIO_MASTER_PROJECT)
 		include_directories("${CMAKE_SOURCE_DIR}/clara")
 
 		# HTTP parser
-		IF (RESTINIO_USE_EXTERNAL_HTTP_PARSER)
-			find_package(http-parser REQUIRED)
-		ELSE ()
-			add_subdirectory(nodejs/http_parser)
-		ENDIF ()
+		add_subdirectory(nodejs/http_parser)
 
 		# fmtlib
 		add_subdirectory(fmt)
