From 2fb64a2af2e3519feee18959e260ec90bcbd1f56 Mon Sep 17 00:00:00 2001
From: Adrien Beraud <adrien.beraud@savoirfairelinux.com>
Date: Fri, 6 Apr 2018 22:39:08 -0400
Subject: [PATCH] build: don't fetch git tag if openbsd directory exists

---
 update.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/update.sh b/update.sh
index 9e9ace6..b064154 100755
--- a/update.sh
+++ b/update.sh
@@ -11,11 +11,11 @@ if [ ! -d openbsd ]; then
 	else
 		git clone $LIBRESSL_GIT/openbsd
 	fi
+	(cd openbsd
+	 git fetch
+	 git checkout $openbsd_branch
+	 git pull --rebase)
 fi
-(cd openbsd
- git fetch
- git checkout $openbsd_branch
- git pull --rebase)
 
 # setup source paths
 CWD=`pwd`
-- 
2.14.1

