From b381d7010e03ada41f36f529468b1ee553a9c765 Mon Sep 17 00:00:00 2001
From: Sebastien Blin <sebastien.blin@savoirfairelinux.com>
Date: Fri, 9 Mar 2018 12:07:17 -0500
Subject: [PATCH] fix redefinition of getpagesize

---
 crypto/compat/getpagesize.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/compat/getpagesize.c b/crypto/compat/getpagesize.c
index 098efa9..79fcc56 100644
--- a/crypto/compat/getpagesize.c
+++ b/crypto/compat/getpagesize.c
@@ -6,6 +6,7 @@
 #include <windows.h>
 #endif

+#ifndef __ANDROID__
 int
 getpagesize(void) {
 #ifdef _MSC_VER
@@ -16,3 +17,4 @@ getpagesize(void) {
 	return sysconf(_SC_PAGESIZE);
 #endif
 }
+#endif
--
2.14.3
