 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d14bc98..e5ad5c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,12 @@ option(DHTNET_TESTABLE "Enable API for tests" ON)
 option(BUILD_TOOLS "Build tools" ON)

 if (NOT MSVC)
+    OPTION(LIBJAMI_TESTABLE "Enable API for tests" ON)
+
+    IF(LIBJAMI_TESTABLE)
+        ADD_DEFINITIONS(-DLIBJAMI_TESTABLE)
+    ENDIF(LIBJAMI_TESTABLE)
+
     find_package (PkgConfig REQUIRED)

     find_package(msgpack-cxx CONFIG)
