From c9578eb29da666b71d6aab6fac79c26631bec603 Mon Sep 17 00:00:00 2001
From: philippegorley <gorley.philippe@gmail.com>
Date: Tue, 25 Jul 2017 13:00:19 -0400
Subject: [PATCH] use internal log2/log2f

---
 configure | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 9b5789a..a44d8d6 100755
--- a/configure
+++ b/configure
@@ -1926,8 +1926,6 @@ MATH_FUNCS="
     ldexpf
     llrint
     llrintf
-    log2
-    log2f
     log10f
     lrint
     lrintf
@@ -6835,6 +6833,9 @@ print_config CONFIG_ "$config_files" $CONFIG_LIST       \
                                      $CONFIG_EXTRA      \
                                      $ALL_COMPONENTS    \
 
+echo "#define HAVE_LOG2 0" >> $TMPH
+echo "#define HAVE_LOG2F 0" >> $TMPH
+
 echo "#endif /* FFMPEG_CONFIG_H */" >> $TMPH
 echo "endif # FFMPEG_CONFIG_MAK" >> config.mak
 
-- 
2.7.4

