--- a/configure	2017-10-03 15:05:34.368369231 -0400
+++ b/configure	2017-10-03 15:06:06.320341666 -0400
@@ -4688,9 +4688,9 @@
     check_cflags -std=c99
 fi

-check_cc -D_FILE_OFFSET_BITS=64 <<EOF && add_cppflags -D_FILE_OFFSET_BITS=64
-#include <stdlib.h>
-EOF
+#check_cc -D_FILE_OFFSET_BITS=64 <<EOF && add_cppflags -D_FILE_OFFSET_BITS=64
+##include <stdlib.h>
+#EOF
 check_cc -D_LARGEFILE_SOURCE <<EOF && add_cppflags -D_LARGEFILE_SOURCE
 #include <stdlib.h>
 EOF
