From 5a8bde32280b733a677577c046fc745dfbda1f8f Mon Sep 17 00:00:00 2001
From: philippegorley <philippe.gorley@savoirfairelinux.com>
Date: Tue, 20 Mar 2018 17:08:20 -0400
Subject: [PATCH] android_file_offset

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 5e38bda..4a8fd61 100755
--- a/configure
+++ b/configure
@@ -4865,8 +4865,8 @@ test_cflags_cc -std=c11 ctype.h "__STDC_VERSION__ >= 201112L" &&
     add_cflags -std=c11 ||
     check_cflags -std=c99

-check_cppflags -D_FILE_OFFSET_BITS=64
-check_cppflags -D_LARGEFILE_SOURCE
+#check_cppflags -D_FILE_OFFSET_BITS=64
+#check_cppflags -D_LARGEFILE_SOURCE

 add_host_cppflags -D_ISOC99_SOURCE
 check_host_cflags -std=c99
--
2.7.4
