From c557a6211f5a29d89fc2ab561e0fb3d8878fb6ad Mon Sep 17 00:00:00 2001
From: philippegorley <gorley.philippe@gmail.com>
Date: Mon, 17 Dec 2018 15:27:57 -0500
Subject: [PATCH] remove mjpeg log

---
 mjpegdec.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/mjpegdec.c b/mjpegdec.c
index d1dca84d36..4a26c23cd4 100644
--- a/mjpegdec.c
+++ b/mjpegdec.c
@@ -2294,9 +2294,7 @@ int ff_mjpeg_decode_frame(AVCodecContext *avctx, void *data, int *got_frame,
                    "restart marker: %d\n", start_code & 0x0f);
             /* APP fields */
         } else if (start_code >= APP0 && start_code <= APP15) {
-            if ((ret = mjpeg_decode_app(s)) < 0)
-                av_log(avctx, AV_LOG_ERROR, "unable to decode APP fields: %s\n",
-                       av_err2str(ret));
+            ret = mjpeg_decode_app(s);
             /* Comment */
         } else if (start_code == COM) {
             ret = mjpeg_decode_com(s);
--
2.17.1
