--- a/configure
+++ b/configure
@@ -1486,7 +1486,7 @@ require_cpp_condition(){
 require_pkg_config(){
     log require_pkg_config "$@"
     pkg_version="$2"
-    check_pkg_config "$@" || die "ERROR: $pkg_version not found using pkg-config$pkg_config_fail_message"
+    #check_pkg_config "$@" || die "ERROR: $pkg_version not found using pkg-config$pkg_config_fail_message"
 }
 
 hostcc_e(){
-- 
2.10.2.windows.1

