--- a/CMakeLists.txt
+++ b/CMakeLists.txt.patched
@@ -39,12 +39,7 @@ set(CRYPTOPP_DATA_DIR "" CACHE PATH "Crypto++ test data directory")
 # Always 1 ahead in Master. Also see http://groups.google.com/forum/#!topic/cryptopp-users/SFhqLDTQPG4
 set(LIB_VER ${cryptopp_VERSION_MAJOR}${cryptopp_VERSION_MINOR}${cryptopp_VERSION_PATCH})
 
-# Only set when cross-compiling, http://www.vtk.org/Wiki/CMake_Cross_Compiling
-if (NOT (CMAKE_SYSTEM_VERSION AND CMAKE_SYSTEM_PROCESSOR))
-	set(CRYPTOPP_CROSS_COMPILE 1)
-else()
-	set(CRYPTOPP_CROSS_COMPILE 0)
-endif()
+set(CRYPTOPP_CROSS_COMPILE 1)
 
 # Don't use RPATH's. The resulting binary could fail a security audit.
 if (NOT CMAKE_VERSION VERSION_LESS 2.8.12)
