--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,6 @@
+# Visual Studio
+*.VC.db
+
 ####################
 ## Crypto++ specific
 ####################
@@ -268,4 +271,4 @@ pip-log.txt
 *.local
 
 # Dependency file
-*.d
+*.d
\ No newline at end of file
--- a/cryptest.vcxproj
+++ b/cryptest.vcxproj
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup Label="ProjectConfigurations">
     <ProjectConfiguration Include="Debug|Win32">
       <Configuration>Debug</Configuration>
@@ -37,55 +37,64 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{9EAFA456-89B4-4879-AD4F-C2C341184CF5}</ProjectGuid>
     <RootNamespace>cryptest</RootNamespace>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v140</PlatformToolset>
     <UseOfMfc>false</UseOfMfc>
     <CharacterSet>MultiByte</CharacterSet>
-    <WholeProgramOptimization>true</WholeProgramOptimization>
+    <WholeProgramOptimization>false</WholeProgramOptimization>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">
@@ -116,23 +125,39 @@
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup>
-    <_ProjectFileVersion>10.0.40219.1</_ProjectFileVersion>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">$(Platform)\Output\$(Configuration)\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='Release|x64'">$(Platform)\Output\$(Configuration)\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='Release|x64'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">$(Platform)\DLL_Output\Release\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">$(Platform)\DLL_Output\Release\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">$(Platform)\DLL_Output\Debug\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">$(Platform)\DLL_Output\Debug\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">$(Platform)\Output\$(Configuration)\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
-    <OutDir Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">$(Platform)\Output\$(Configuration)\</OutDir>
-    <IntDir Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+    <_ProjectFileVersion>14.0.25420.1</_ProjectFileVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <OutDir>$(Platform)\Output\$(Configuration)\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <OutDir>$(Platform)\Output\$(Configuration)\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">
+    <OutDir>$(Platform)\DLL_Output\Release\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">
+    <OutDir>$(Platform)\DLL_Output\Release\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">
+    <OutDir>$(Platform)\DLL_Output\Debug\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">
+    <OutDir>$(Platform)\DLL_Output\Debug\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <OutDir>$(Platform)\Output\$(Configuration)\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <OutDir>$(Platform)\Output\$(Configuration)\</OutDir>
+    <IntDir>$(Platform)\$(ProjectName)\$(Configuration)\</IntDir>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
     <ClCompile>
@@ -144,8 +169,7 @@
       <StringPooling>true</StringPooling>
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
       <FunctionLevelLinking>true</FunctionLevelLinking>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -157,7 +181,7 @@
     <Link>
       <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <OptimizeReferences>true</OptimizeReferences>
@@ -175,8 +199,7 @@
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <EnableEnhancedInstructionSet>NotSet</EnableEnhancedInstructionSet>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -188,7 +211,7 @@
     <Link>
       <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <OptimizeReferences>true</OptimizeReferences>
@@ -206,8 +229,7 @@
       <StringPooling>true</StringPooling>
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
       <FunctionLevelLinking>true</FunctionLevelLinking>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -225,9 +247,9 @@ echo unless it undergoes FIPS validation.
 </Command>
     </PreLinkEvent>
     <Link>
-      <AdditionalDependencies>cryptlib.lib;cryptopp.lib;Ws2_32.lib</AdditionalDependencies>
+      <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Release</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Release;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <OptimizeReferences>true</OptimizeReferences>
@@ -244,8 +266,7 @@ echo unless it undergoes FIPS validation.
       <StringPooling>true</StringPooling>
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
       <FunctionLevelLinking>true</FunctionLevelLinking>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -263,9 +284,9 @@ echo unless it undergoes FIPS validation.
 </Command>
     </PreLinkEvent>
     <Link>
-      <AdditionalDependencies>cryptlib.lib;cryptopp.lib;Ws2_32.lib</AdditionalDependencies>
+      <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Release</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Release;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <OptimizeReferences>true</OptimizeReferences>
@@ -280,8 +301,7 @@ echo unless it undergoes FIPS validation.
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <PreprocessorDefinitions>_DEBUG;CRYPTOPP_IMPORTS;WIN32;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -299,9 +319,9 @@ echo unless it undergoes FIPS validation.
 </Command>
     </PreLinkEvent>
     <Link>
-      <AdditionalDependencies>cryptlib.lib;cryptopp.lib;Ws2_32.lib</AdditionalDependencies>
+      <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Debug</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Debug;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
     </Link>
@@ -313,8 +333,7 @@ echo unless it undergoes FIPS validation.
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <PreprocessorDefinitions>_DEBUG;CRYPTOPP_IMPORTS;WIN32;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -332,9 +351,9 @@ echo unless it undergoes FIPS validation.
 </Command>
     </PreLinkEvent>
     <Link>
-      <AdditionalDependencies>cryptlib.lib;cryptopp.lib;Ws2_32.lib</AdditionalDependencies>
+      <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Debug</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\DLL_Output\Debug;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <TargetMachine>MachineX64</TargetMachine>
@@ -347,8 +366,7 @@ echo unless it undergoes FIPS validation.
       <PreprocessorDefinitions>_DEBUG;_CONSOLE;WIN32;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
       <EnableEnhancedInstructionSet>StreamingSIMDExtensions</EnableEnhancedInstructionSet>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -360,7 +378,7 @@ echo unless it undergoes FIPS validation.
     <Link>
       <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
     </Link>
@@ -371,8 +389,7 @@ echo unless it undergoes FIPS validation.
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <PreprocessorDefinitions>_DEBUG;_CONSOLE;WIN32;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <PrecompiledHeader>
-      </PrecompiledHeader>
+      <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -384,285 +401,142 @@ echo unless it undergoes FIPS validation.
     <Link>
       <AdditionalDependencies>cryptlib.lib;Ws2_32.lib</AdditionalDependencies>
       <SuppressStartupBanner>true</SuppressStartupBanner>
-      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories>$(Platform)\Output\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
       <TargetMachine>MachineX64</TargetMachine>
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
-    <None Include="TestVectors\aes.txt" />
-    <None Include="TestVectors\all.txt" />
-    <None Include="TestVectors\blake2.txt" />
-    <None Include="TestVectors\blake2b.txt" />
-    <None Include="TestVectors\blake2s.txt" />
-    <None Include="TestVectors\camellia.txt" />
-    <None Include="TestVectors\ccm.txt" />
-    <None Include="TestVectors\chacha.txt" />
-    <None Include="TestVectors\cmac.txt" />
-    <None Include="TestVectors\dlies.txt" />
-    <None Include="TestVectors\dsa.txt" />
-    <None Include="TestVectors\dsa_1363.txt" />
-    <None Include="TestVectors\eax.txt" />
-    <None Include="TestVectors\esign.txt" />
-    <None Include="TestVectors\gcm.txt" />
-    <None Include="TestVectors\hkdf.txt" />
-    <None Include="TestVectors\hmac.txt" />
-    <None Include="TestVectors\mars.txt" />
-    <None Include="TestVectors\nr.txt" />
-    <None Include="TestVectors\panama.txt" />
-    <None Include="TestVectors\Readme.txt" />
-    <None Include="TestVectors\rsa_oaep.txt" />
-    <None Include="TestVectors\rsa_pkcs1_1_5.txt" />
-    <None Include="TestVectors\rsa_pss.txt" />
-    <None Include="TestVectors\rw.txt" />
-    <None Include="TestVectors\salsa.txt" />
-    <None Include="TestVectors\seal.txt" />
-    <None Include="TestVectors\seed.txt" />
-    <None Include="TestVectors\sha.txt" />
-    <None Include="TestVectors\sha3.txt" />
-    <None Include="TestVectors\shacal2.txt" />
-    <None Include="TestVectors\sosemanuk.txt" />
-    <None Include="TestVectors\tea.txt" />
-    <None Include="TestVectors\ttmac.txt" />
-    <None Include="TestVectors\vmac.txt" />
-    <None Include="TestVectors\wake.txt" />
-    <None Include="TestVectors\whrlpool.txt" />
-    <None Include="TestData\3desval.dat" />
-    <None Include="TestData\3wayval.dat" />
-    <None Include="TestData\camellia.dat" />
-    <None Include="TestData\cast128v.dat" />
-    <None Include="TestData\cast256v.dat" />
-    <None Include="TestData\descert.dat" />
-    <None Include="TestData\dh1024.dat" />
-    <None Include="TestData\dh2048.dat" />
-    <None Include="TestData\dlie1024.dat" />
-    <None Include="TestData\dlie2048.dat" />
-    <None Include="TestData\dsa1024.dat" />
-    <None Include="TestData\dsa1024b.dat" />
-    <None Include="TestData\dsa512.dat" />
-    <None Include="TestData\elgc1024.dat" />
-    <None Include="TestData\esig1023.dat" />
-    <None Include="TestData\esig1536.dat" />
-    <None Include="TestData\esig2046.dat" />
-    <None Include="TestData\fhmqv160.dat" />
-    <None Include="TestData\fhmqv256.dat" />
-    <None Include="TestData\fhmqv384.dat" />
-    <None Include="TestData\fhmqv512.dat" />
-    <None Include="TestData\gostval.dat" />
-    <None Include="TestData\hmqv160.dat" />
-    <None Include="TestData\hmqv256.dat" />
-    <None Include="TestData\hmqv384.dat" />
-    <None Include="TestData\hmqv512.dat" />
-    <None Include="TestData\ideaval.dat" />
-    <None Include="TestData\luc1024.dat" />
-    <None Include="TestData\luc2048.dat" />
-    <None Include="TestData\lucc1024.dat" />
-    <None Include="TestData\lucc512.dat" />
-    <None Include="TestData\lucd1024.dat" />
-    <None Include="TestData\lucd512.dat" />
-    <None Include="TestData\lucs1024.dat" />
-    <None Include="TestData\lucs512.dat" />
-    <None Include="TestData\marsval.dat" />
-    <None Include="TestData\mqv1024.dat" />
-    <None Include="TestData\mqv2048.dat" />
-    <None Include="TestData\nr1024.dat" />
-    <None Include="TestData\nr2048.dat" />
-    <None Include="TestData\rabi1024.dat" />
-    <None Include="TestData\rabi2048.dat" />
-    <None Include="TestData\rc2val.dat" />
-    <None Include="TestData\rc5val.dat" />
-    <None Include="TestData\rc6val.dat" />
-    <None Include="TestData\rijndael.dat" />
-    <None Include="TestData\rsa1024.dat" />
-    <None Include="TestData\rsa2048.dat" />
-    <None Include="TestData\rsa400pb.dat" />
-    <None Include="TestData\rsa400pv.dat" />
-    <None Include="TestData\rsa512a.dat" />
-    <None Include="TestData\rw1024.dat" />
-    <None Include="TestData\rw2048.dat" />
-    <None Include="TestData\saferval.dat" />
-    <None Include="TestData\serpentv.dat" />
-    <None Include="TestData\shacal2v.dat" />
-    <None Include="TestData\sharkval.dat" />
-    <None Include="TestData\skipjack.dat" />
-    <None Include="TestData\squareva.dat" />
-    <None Include="TestData\twofishv.dat" />
-    <None Include="TestData\usage.dat" />
-    <None Include="TestData\xtrdh171.dat" />
-    <None Include="TestData\xtrdh342.dat" />
+    <Text Include="TestVectors\aes.txt" />
+    <Text Include="TestVectors\all.txt" />
+    <Text Include="TestVectors\blake2.txt" />
+    <Text Include="TestVectors\blake2b.txt" />
+    <Text Include="TestVectors\blake2s.txt" />
+    <Text Include="TestVectors\camellia.txt" />
+    <Text Include="TestVectors\ccm.txt" />
+    <Text Include="TestVectors\chacha.txt" />
+    <Text Include="TestVectors\cmac.txt" />
+    <Text Include="TestVectors\dlies.txt" />
+    <Text Include="TestVectors\dsa.txt" />
+    <Text Include="TestVectors\dsa_1363.txt" />
+    <Text Include="TestVectors\eax.txt" />
+    <Text Include="TestVectors\esign.txt" />
+    <Text Include="TestVectors\gcm.txt" />
+    <Text Include="TestVectors\hkdf.txt" />
+    <Text Include="TestVectors\hmac.txt" />
+    <Text Include="TestVectors\mars.txt" />
+    <Text Include="TestVectors\nr.txt" />
+    <Text Include="TestVectors\panama.txt" />
+    <Text Include="TestVectors\Readme.txt" />
+    <Text Include="TestVectors\rsa_oaep.txt" />
+    <Text Include="TestVectors\rsa_pkcs1_1_5.txt" />
+    <Text Include="TestVectors\rsa_pss.txt" />
+    <Text Include="TestVectors\rw.txt" />
+    <Text Include="TestVectors\salsa.txt" />
+    <Text Include="TestVectors\seal.txt" />
+    <Text Include="TestVectors\seed.txt" />
+    <Text Include="TestVectors\sha.txt" />
+    <Text Include="TestVectors\sha3.txt" />
+    <Text Include="TestVectors\shacal2.txt" />
+    <Text Include="TestVectors\sosemanuk.txt" />
+    <Text Include="TestVectors\tea.txt" />
+    <Text Include="TestVectors\ttmac.txt" />
+    <Text Include="TestVectors\vmac.txt" />
+    <Text Include="TestVectors\wake.txt" />
+    <Text Include="TestVectors\whrlpool.txt" />
   </ItemGroup>
   <ItemGroup>
     <ClCompile Include="adhoc.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="bench1.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="bench2.cpp" />
     <ClCompile Include="datatest.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="dlltest.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="fipsalgt.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="regtest.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="test.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="validat1.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="validat2.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
     <ClCompile Include="validat3.cpp">
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">MaxSpeed</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='DLL-Import Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Disabled</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">MinSpace</Optimization>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(Configuration)|$(Platform)'=='Release|x64'">%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
   </ItemGroup>
   <ItemGroup>
@@ -670,7 +544,77 @@ echo unless it undergoes FIPS validation.
     <ClInclude Include="factory.h" />
     <ClInclude Include="validate.h" />
   </ItemGroup>
+  <ItemGroup>
+    <None Include="TestData\3desval.dat" />
+    <None Include="TestData\3wayval.dat" />
+    <None Include="TestData\camellia.dat" />
+    <None Include="TestData\cast128v.dat" />
+    <None Include="TestData\cast256v.dat" />
+    <None Include="TestData\descert.dat" />
+    <None Include="TestData\dh1024.dat" />
+    <None Include="TestData\dh2048.dat" />
+    <None Include="TestData\dlie1024.dat" />
+    <None Include="TestData\dlie2048.dat" />
+    <None Include="TestData\dsa1024.dat" />
+    <None Include="TestData\dsa1024b.dat" />
+    <None Include="TestData\dsa512.dat" />
+    <None Include="TestData\elgc1024.dat" />
+    <None Include="TestData\esig1023.dat" />
+    <None Include="TestData\esig1536.dat" />
+    <None Include="TestData\esig2046.dat" />
+    <None Include="TestData\fhmqv160.dat" />
+    <None Include="TestData\fhmqv256.dat" />
+    <None Include="TestData\fhmqv384.dat" />
+    <None Include="TestData\fhmqv512.dat" />
+    <None Include="TestData\gostval.dat" />
+    <None Include="TestData\hmqv160.dat" />
+    <None Include="TestData\hmqv256.dat" />
+    <None Include="TestData\hmqv384.dat" />
+    <None Include="TestData\ideaval.dat" />
+    <None Include="TestData\luc1024.dat" />
+    <None Include="TestData\luc2048.dat" />
+    <None Include="TestData\lucc1024.dat" />
+    <None Include="TestData\lucc512.dat" />
+    <None Include="TestData\lucd1024.dat" />
+    <None Include="TestData\lucd512.dat" />
+    <None Include="TestData\lucs1024.dat" />
+    <None Include="TestData\lucs512.dat" />
+    <None Include="TestData\marsval.dat" />
+    <None Include="TestData\mqv1024.dat" />
+    <None Include="TestData\mqv2048.dat" />
+    <None Include="TestData\nr1024.dat" />
+    <None Include="TestData\nr2048.dat" />
+    <None Include="TestData\rabi1024.dat" />
+    <None Include="TestData\rabi2048.dat" />
+    <None Include="TestData\rc2val.dat" />
+    <None Include="TestData\rc5val.dat" />
+    <None Include="TestData\rc6val.dat" />
+    <None Include="TestData\rijndael.dat" />
+    <None Include="TestData\rsa1024.dat" />
+    <None Include="TestData\rsa2048.dat" />
+    <None Include="TestData\rsa400pb.dat" />
+    <None Include="TestData\rsa400pv.dat" />
+    <None Include="TestData\rsa512a.dat" />
+    <None Include="TestData\rw1024.dat" />
+    <None Include="TestData\rw2048.dat" />
+    <None Include="TestData\saferval.dat" />
+    <None Include="TestData\serpentv.dat" />
+    <None Include="TestData\shacal2v.dat" />
+    <None Include="TestData\sharkval.dat" />
+    <None Include="TestData\skipjack.dat" />
+    <None Include="TestData\squareva.dat" />
+    <None Include="TestData\twofishv.dat" />
+    <None Include="TestData\usage.dat" />
+    <None Include="TestData\xtrdh171.dat" />
+    <None Include="TestData\xtrdh342.dat" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="cryptlib.vcxproj">
+      <Project>{3423ec9a-52e4-4a4d-9753-edebc38785ef}</Project>
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+    </ProjectReference>
+  </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
\ No newline at end of file
--- a/hrtimer.cpp
+++ b/hrtimer.cpp
@@ -26,7 +26,7 @@
 #include <unistd.h>
 #endif
 
-#include "trap.h"
+#include <assert.h>
 
 NAMESPACE_BEGIN(CryptoPP)
 
@@ -53,7 +53,7 @@ double TimerBase::ConvertTo(TimerWord t, Unit unit)
 	static unsigned long unitsPerSecondTable[] = {1, 1000, 1000*1000, 1000*1000*1000};
 
 	// When 'unit' is an enum 'Unit', a Clang warning is generated.
-	CRYPTOPP_ASSERT(static_cast<unsigned int>(unit) < COUNTOF(unitsPerSecondTable));
+	assert(static_cast<unsigned int>(unit) < COUNTOF(unitsPerSecondTable));
 	return (double)CRYPTOPP_VC6_INT64 t * unitsPerSecondTable[unit] / CRYPTOPP_VC6_INT64 TicksPerSecond();
 }
 
@@ -83,7 +83,7 @@ double TimerBase::ElapsedTimeAsDouble()
 unsigned long TimerBase::ElapsedTime()
 {
 	double elapsed = ElapsedTimeAsDouble();
-	CRYPTOPP_ASSERT(elapsed <= (double)ULONG_MAX);
+	assert(elapsed <= (double)ULONG_MAX);
 	return (unsigned long)elapsed;
 }
 
@@ -121,7 +121,7 @@ TimerWord Timer::TicksPerSecond()
 TimerWord ThreadUserTimer::GetCurrentTimerValue()
 {
 #if defined(CRYPTOPP_WIN32_AVAILABLE) && defined(THREAD_TIMER_AVAILABLE)
-	static bool getCurrentThreadImplemented = true;
+	/*static bool getCurrentThreadImplemented = true;
 	if (getCurrentThreadImplemented)
 	{
 		FILETIME now, ignored;
@@ -137,14 +137,14 @@ TimerWord ThreadUserTimer::GetCurrentTimerValue()
 		}
 		return now.dwLowDateTime + ((TimerWord)now.dwHighDateTime << 32);
 	}
-GetCurrentThreadNotImplemented:
+GetCurrentThreadNotImplemented:*/
 	return (TimerWord)clock() * (10*1000*1000 / CLOCKS_PER_SEC);
 #elif defined(CRYPTOPP_WIN32_AVAILABLE) && !defined(THREAD_TIMER_AVAILABLE)
 	LARGE_INTEGER now;
 	if (!QueryPerformanceCounter(&now))
 	{
 		const DWORD lastError = GetLastError();
-		throw Exception(Exception::OTHER_ERROR, "ThreadUserTimer: QueryPerformanceCounter failed with error " + IntToString(lastError));
+		throw Exception(Exception::OTHER_ERROR, "ThreadUserTimer: QueryPerformanceCounter failed with error " + IntToString(lastError));		
 	}
 	return now.QuadPart;
 #elif defined(CRYPTOPP_UNIX_AVAILABLE)
--- a/wait.cpp
+++ b/wait.cpp
@@ -63,7 +63,7 @@ void WaitObjectContainer::Clear()
 # ifdef CRYPTOPP_MSAN
 	__msan_unpoison(&m_readfds, sizeof(m_readfds));
 	__msan_unpoison(&m_writefds, sizeof(m_writefds));
-# endif
+# endif	
 #endif
 	m_noWait = false;
 	m_firstEventTime = 0;
@@ -137,7 +137,7 @@ WaitObjectContainer::~WaitObjectContainer()
 		if (!m_threads.empty())
 		{
 			HANDLE threadHandles[MAXIMUM_WAIT_OBJECTS] = {0};
-
+			
 			unsigned int i;
 			for (i=0; i<m_threads.size(); i++)
 			{
@@ -151,36 +151,36 @@ WaitObjectContainer::~WaitObjectContainer()
 				threadHandles[i] = thread.threadHandle;
 			}
 
-			BOOL bResult = PulseEvent(m_startWaiting);
-			CRYPTOPP_ASSERT(bResult != 0); CRYPTOPP_UNUSED(bResult);
-
+			BOOL bResult = false;//PulseEvent(m_startWaiting);
+			assert(bResult != 0); CRYPTOPP_UNUSED(bResult);
+	
 			// Enterprise Analysis warning
 #if defined(USE_WINDOWS8_API)
 			DWORD dwResult = ::WaitForMultipleObjectsEx((DWORD)m_threads.size(), threadHandles, TRUE, INFINITE, FALSE);
-			CRYPTOPP_ASSERT((dwResult >= WAIT_OBJECT_0) && (dwResult < (DWORD)m_threads.size()));
+			assert((dwResult >= WAIT_OBJECT_0) && (dwResult < (DWORD)m_threads.size()));
 #else
 			DWORD dwResult = ::WaitForMultipleObjects((DWORD)m_threads.size(), threadHandles, TRUE, INFINITE);
-			CRYPTOPP_ASSERT((dwResult >= WAIT_OBJECT_0) && (dwResult < (DWORD)m_threads.size()));
+			assert((dwResult >= WAIT_OBJECT_0) && (dwResult < (DWORD)m_threads.size()));
 #endif
 
 			for (i=0; i<m_threads.size(); i++)
 			{
 				// Enterprise Analysis warning
 				if (!threadHandles[i]) continue;
-
+									
 				bResult = CloseHandle(threadHandles[i]);
-				CRYPTOPP_ASSERT(bResult != 0);
+				assert(bResult != 0);
 			}
 
 			bResult = CloseHandle(m_startWaiting);
-			CRYPTOPP_ASSERT(bResult != 0);
+			assert(bResult != 0);
 			bResult = CloseHandle(m_stopWaiting);
-			CRYPTOPP_ASSERT(bResult != 0);
+			assert(bResult != 0);
 		}
 	}
 	catch (const Exception&)
 	{
-		CRYPTOPP_ASSERT(0);
+		assert(0);
 	}
 }
 
@@ -201,12 +201,12 @@ DWORD WINAPI WaitingThread(LPVOID lParam)
 		thread.waitingToWait = true;
 #if defined(USE_WINDOWS8_API)
 		DWORD result = ::WaitForSingleObjectEx(thread.startWaiting, INFINITE, FALSE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #else
 		DWORD result = ::WaitForSingleObject(thread.startWaiting, INFINITE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #endif
-
+		
 		thread.waitingToWait = false;
 		if (thread.terminate)
 			break;
@@ -219,10 +219,10 @@ DWORD WINAPI WaitingThread(LPVOID lParam)
 
 #if defined(USE_WINDOWS8_API)
 		result = ::WaitForMultipleObjectsEx((DWORD)handles.size(), &handles[0], FALSE, INFINITE, FALSE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #else
 		result = ::WaitForMultipleObjects((DWORD)handles.size(), &handles[0], FALSE, INFINITE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #endif
 
 		if (result == WAIT_OBJECT_0)
@@ -230,7 +230,7 @@ DWORD WINAPI WaitingThread(LPVOID lParam)
 		SetEvent(thread.stopWaiting);
 		if (!(result > WAIT_OBJECT_0 && result < WAIT_OBJECT_0 + handles.size()))
 		{
-			CRYPTOPP_ASSERT(!"error in WaitingThread");	// break here so we can see which thread has an error
+			assert(!"error in WaitingThread");	// break here so we can see which thread has an error
 			*thread.error = ::GetLastError();
 		}
 	}
@@ -254,7 +254,7 @@ void WaitObjectContainer::CreateThreads(unsigned int count)
 		{
 			// Enterprise Analysis warning
 			if(!m_threads[i]) continue;
-
+	
 			m_threads[i] = new WaitingThreadData;
 			WaitingThreadData &thread = *m_threads[i];
 			thread.terminate = false;
@@ -304,7 +304,7 @@ bool WaitObjectContainer::Wait(unsigned long milliseconds)
 			throw Err("WaitObjectContainer: number of wait objects exceeds limit");
 		CreateThreads(nThreads);
 		DWORD error = S_OK;
-
+		
 		for (unsigned int i=0; i<m_threads.size(); i++)
 		{
 			// Enterprise Analysis warning
@@ -324,14 +324,14 @@ bool WaitObjectContainer::Wait(unsigned long milliseconds)
 		}
 
 		ResetEvent(m_stopWaiting);
-		PulseEvent(m_startWaiting);
+		//PulseEvent(m_startWaiting);
 
 #if defined(USE_WINDOWS8_API)
 		DWORD result = ::WaitForSingleObjectEx(m_stopWaiting, milliseconds, FALSE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #else
 		DWORD result = ::WaitForSingleObject(m_stopWaiting, milliseconds);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #endif
 
 		if (result == WAIT_OBJECT_0)
@@ -359,10 +359,10 @@ bool WaitObjectContainer::Wait(unsigned long milliseconds)
 #endif
 #if defined(USE_WINDOWS8_API)
 		DWORD result = ::WaitForMultipleObjectsEx((DWORD)m_handles.size(), &m_handles[0], FALSE, milliseconds, FALSE);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #else
 		DWORD result = ::WaitForMultipleObjects((DWORD)m_handles.size(), &m_handles[0], FALSE, milliseconds);
-		CRYPTOPP_ASSERT(result != WAIT_FAILED);
+		assert(result != WAIT_FAILED);
 #endif
 #if TRACE_WAIT
 		if (milliseconds > 0)
-- 
2.8.1.windows.1

