--- /dev/null
+++ b/asio/msvc/asio.vcxproj
@@ -0,0 +1,567 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>15.0</VCProjectVersion>
+    <ProjectGuid>{08A80A1E-58CD-43F6-86B2-DAD475D573D0}</ProjectGuid>
+    <RootNamespace>asio</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0.17763.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseDebugLibraries>true</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
+    <CharacterSet>Unicode</CharacterSet>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseDebugLibraries>false</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
+    <WholeProgramOptimization>true</WholeProgramOptimization>
+    <CharacterSet>Unicode</CharacterSet>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseDebugLibraries>true</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
+    <CharacterSet>Unicode</CharacterSet>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <UseDebugLibraries>false</UseDebugLibraries>
+    <PlatformToolset>v141</PlatformToolset>
+    <WholeProgramOptimization>true</WholeProgramOptimization>
+    <CharacterSet>Unicode</CharacterSet>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <ImportGroup Label="ExtensionSettings">
+  </ImportGroup>
+  <ImportGroup Label="Shared">
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup />
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
+    <ClCompile>
+      <WarningLevel>Level3</WarningLevel>
+      <Optimization>Disabled</Optimization>
+      <SDLCheck>true</SDLCheck>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\include;$(ProjectDir)..\..\..\openssl\inc32;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>_WIN32_WINNT=0x0A00;ASIO_STANDALONE;ASIO_SEPARATE_COMPILATION;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+    <Link>
+      <AdditionalLibraryDirectories>$(ProjectDir)..\..\..\openssl\out32dll;</AdditionalLibraryDirectories>
+    </Link>
+    <Link>
+      <AdditionalDependencies>libeay32.lib;ssleay32.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <ClCompile>
+      <WarningLevel>Level3</WarningLevel>
+      <Optimization>Disabled</Optimization>
+      <SDLCheck>true</SDLCheck>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\include;$(ProjectDir)..\..\..\openssl\inc32;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>_WIN32_WINNT=0x0A00;ASIO_STANDALONE;ASIO_SEPARATE_COMPILATION;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+    <Link>
+      <AdditionalLibraryDirectories>$(ProjectDir)..\..\..\openssl\out32dll;</AdditionalLibraryDirectories>
+    </Link>
+    <Link>
+      <AdditionalDependencies>libeay32.lib;ssleay32.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <ClCompile>
+      <WarningLevel>Level3</WarningLevel>
+      <Optimization>MaxSpeed</Optimization>
+      <FunctionLevelLinking>true</FunctionLevelLinking>
+      <IntrinsicFunctions>true</IntrinsicFunctions>
+      <SDLCheck>true</SDLCheck>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\include;$(ProjectDir)..\..\..\openssl\inc32;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>_WIN32_WINNT=0x0A00;ASIO_STANDALONE;ASIO_SEPARATE_COMPILATION;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+    <Link>
+      <EnableCOMDATFolding>true</EnableCOMDATFolding>
+      <OptimizeReferences>true</OptimizeReferences>
+      <AdditionalLibraryDirectories>$(ProjectDir)..\..\..\openssl\out32dll;</AdditionalLibraryDirectories>
+      <AdditionalDependencies>libeay32.lib;ssleay32.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <ClCompile>
+      <WarningLevel>Level3</WarningLevel>
+      <Optimization>MaxSpeed</Optimization>
+      <FunctionLevelLinking>true</FunctionLevelLinking>
+      <IntrinsicFunctions>true</IntrinsicFunctions>
+      <SDLCheck>true</SDLCheck>
+      <ConformanceMode>true</ConformanceMode>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\include;$(ProjectDir)..\..\..\openssl\inc32;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>_WIN32_WINNT=0x0A00;ASIO_STANDALONE;ASIO_SEPARATE_COMPILATION;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+    <Link>
+      <EnableCOMDATFolding>true</EnableCOMDATFolding>
+      <OptimizeReferences>true</OptimizeReferences>
+      <AdditionalLibraryDirectories>$(ProjectDir)..\..\..\openssl\out32dll;</AdditionalLibraryDirectories>
+      <AdditionalDependencies>libeay32.lib;ssleay32.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClCompile Include="..\src\asio.cpp" />
+    <ClCompile Include="..\src\asio_ssl.cpp" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\include\asio.hpp" />
+    <ClInclude Include="..\include\asio\associated_allocator.hpp" />
+    <ClInclude Include="..\include\asio\associated_executor.hpp" />
+    <ClInclude Include="..\include\asio\async_result.hpp" />
+    <ClInclude Include="..\include\asio\basic_datagram_socket.hpp" />
+    <ClInclude Include="..\include\asio\basic_deadline_timer.hpp" />
+    <ClInclude Include="..\include\asio\basic_io_object.hpp" />
+    <ClInclude Include="..\include\asio\basic_raw_socket.hpp" />
+    <ClInclude Include="..\include\asio\basic_seq_packet_socket.hpp" />
+    <ClInclude Include="..\include\asio\basic_serial_port.hpp" />
+    <ClInclude Include="..\include\asio\basic_signal_set.hpp" />
+    <ClInclude Include="..\include\asio\basic_socket.hpp" />
+    <ClInclude Include="..\include\asio\basic_socket_acceptor.hpp" />
+    <ClInclude Include="..\include\asio\basic_socket_iostream.hpp" />
+    <ClInclude Include="..\include\asio\basic_socket_streambuf.hpp" />
+    <ClInclude Include="..\include\asio\basic_streambuf.hpp" />
+    <ClInclude Include="..\include\asio\basic_streambuf_fwd.hpp" />
+    <ClInclude Include="..\include\asio\basic_stream_socket.hpp" />
+    <ClInclude Include="..\include\asio\basic_waitable_timer.hpp" />
+    <ClInclude Include="..\include\asio\bind_executor.hpp" />
+    <ClInclude Include="..\include\asio\buffer.hpp" />
+    <ClInclude Include="..\include\asio\buffered_read_stream.hpp" />
+    <ClInclude Include="..\include\asio\buffered_read_stream_fwd.hpp" />
+    <ClInclude Include="..\include\asio\buffered_stream.hpp" />
+    <ClInclude Include="..\include\asio\buffered_stream_fwd.hpp" />
+    <ClInclude Include="..\include\asio\buffered_write_stream.hpp" />
+    <ClInclude Include="..\include\asio\buffered_write_stream_fwd.hpp" />
+    <ClInclude Include="..\include\asio\buffers_iterator.hpp" />
+    <ClInclude Include="..\include\asio\completion_condition.hpp" />
+    <ClInclude Include="..\include\asio\connect.hpp" />
+    <ClInclude Include="..\include\asio\coroutine.hpp" />
+    <ClInclude Include="..\include\asio\datagram_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\deadline_timer.hpp" />
+    <ClInclude Include="..\include\asio\deadline_timer_service.hpp" />
+    <ClInclude Include="..\include\asio\defer.hpp" />
+    <ClInclude Include="..\include\asio\detail\array.hpp" />
+    <ClInclude Include="..\include\asio\detail\array_fwd.hpp" />
+    <ClInclude Include="..\include\asio\detail\assert.hpp" />
+    <ClInclude Include="..\include\asio\detail\atomic_count.hpp" />
+    <ClInclude Include="..\include\asio\detail\base_from_completion_cond.hpp" />
+    <ClInclude Include="..\include\asio\detail\bind_handler.hpp" />
+    <ClInclude Include="..\include\asio\detail\buffered_stream_storage.hpp" />
+    <ClInclude Include="..\include\asio\detail\buffer_resize_guard.hpp" />
+    <ClInclude Include="..\include\asio\detail\buffer_sequence_adapter.hpp" />
+    <ClInclude Include="..\include\asio\detail\call_stack.hpp" />
+    <ClInclude Include="..\include\asio\detail\chrono.hpp" />
+    <ClInclude Include="..\include\asio\detail\chrono_time_traits.hpp" />
+    <ClInclude Include="..\include\asio\detail\completion_handler.hpp" />
+    <ClInclude Include="..\include\asio\detail\concurrency_hint.hpp" />
+    <ClInclude Include="..\include\asio\detail\conditionally_enabled_event.hpp" />
+    <ClInclude Include="..\include\asio\detail\conditionally_enabled_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\config.hpp" />
+    <ClInclude Include="..\include\asio\detail\consuming_buffers.hpp" />
+    <ClInclude Include="..\include\asio\detail\cstddef.hpp" />
+    <ClInclude Include="..\include\asio\detail\cstdint.hpp" />
+    <ClInclude Include="..\include\asio\detail\date_time_fwd.hpp" />
+    <ClInclude Include="..\include\asio\detail\deadline_timer_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\dependent_type.hpp" />
+    <ClInclude Include="..\include\asio\detail\descriptor_ops.hpp" />
+    <ClInclude Include="..\include\asio\detail\descriptor_read_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\descriptor_write_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\dev_poll_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\epoll_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\event.hpp" />
+    <ClInclude Include="..\include\asio\detail\eventfd_select_interrupter.hpp" />
+    <ClInclude Include="..\include\asio\detail\executor_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\fd_set_adapter.hpp" />
+    <ClInclude Include="..\include\asio\detail\fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\functional.hpp" />
+    <ClInclude Include="..\include\asio\detail\future.hpp" />
+    <ClInclude Include="..\include\asio\detail\gcc_arm_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\gcc_hppa_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\gcc_sync_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\gcc_x86_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\global.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_alloc_helpers.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_cont_helpers.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_invoke_helpers.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_tracking.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_type_requirements.hpp" />
+    <ClInclude Include="..\include\asio\detail\handler_work.hpp" />
+    <ClInclude Include="..\include\asio\detail\hash_map.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\dev_poll_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\epoll_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\kqueue_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\select_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\service_registry.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\strand_executor_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\strand_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\winrt_timer_scheduler.hpp" />
+    <ClInclude Include="..\include\asio\detail\impl\win_iocp_io_context.hpp" />
+    <ClInclude Include="..\include\asio\detail\io_control.hpp" />
+    <ClInclude Include="..\include\asio\detail\is_buffer_sequence.hpp" />
+    <ClInclude Include="..\include\asio\detail\is_executor.hpp" />
+    <ClInclude Include="..\include\asio\detail\keyword_tss_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\kqueue_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\limits.hpp" />
+    <ClInclude Include="..\include\asio\detail\local_free_on_block_exit.hpp" />
+    <ClInclude Include="..\include\asio\detail\macos_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\memory.hpp" />
+    <ClInclude Include="..\include\asio\detail\mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\noncopyable.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_event.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_global.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_signal_blocker.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_static_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\null_tss_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\object_pool.hpp" />
+    <ClInclude Include="..\include\asio\detail\old_win_sdk_compat.hpp" />
+    <ClInclude Include="..\include\asio\detail\operation.hpp" />
+    <ClInclude Include="..\include\asio\detail\op_queue.hpp" />
+    <ClInclude Include="..\include\asio\detail\pipe_select_interrupter.hpp" />
+    <ClInclude Include="..\include\asio\detail\pop_options.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_event.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_fd_set_adapter.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_global.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_signal_blocker.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_static_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\posix_tss_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\push_options.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_descriptor_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_null_buffers_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_serial_port_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_accept_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_connect_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_recvfrom_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_recvmsg_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_recv_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_sendto_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_send_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_socket_service_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactive_wait_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactor_fwd.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactor_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\reactor_op_queue.hpp" />
+    <ClInclude Include="..\include\asio\detail\recycling_allocator.hpp" />
+    <ClInclude Include="..\include\asio\detail\regex_fwd.hpp" />
+    <ClInclude Include="..\include\asio\detail\resolver_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\resolver_service_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\resolve_endpoint_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\resolve_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\resolve_query_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\scheduler.hpp" />
+    <ClInclude Include="..\include\asio\detail\scheduler_operation.hpp" />
+    <ClInclude Include="..\include\asio\detail\scheduler_thread_info.hpp" />
+    <ClInclude Include="..\include\asio\detail\scoped_lock.hpp" />
+    <ClInclude Include="..\include\asio\detail\scoped_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\select_interrupter.hpp" />
+    <ClInclude Include="..\include\asio\detail\select_reactor.hpp" />
+    <ClInclude Include="..\include\asio\detail\service_registry.hpp" />
+    <ClInclude Include="..\include\asio\detail\signal_blocker.hpp" />
+    <ClInclude Include="..\include\asio\detail\signal_handler.hpp" />
+    <ClInclude Include="..\include\asio\detail\signal_init.hpp" />
+    <ClInclude Include="..\include\asio\detail\signal_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\signal_set_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\socket_holder.hpp" />
+    <ClInclude Include="..\include\asio\detail\socket_ops.hpp" />
+    <ClInclude Include="..\include\asio\detail\socket_option.hpp" />
+    <ClInclude Include="..\include\asio\detail\socket_select_interrupter.hpp" />
+    <ClInclude Include="..\include\asio\detail\socket_types.hpp" />
+    <ClInclude Include="..\include\asio\detail\solaris_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\static_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_event.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_global.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_static_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\std_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\strand_executor_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\strand_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\string_view.hpp" />
+    <ClInclude Include="..\include\asio\detail\thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\thread_context.hpp" />
+    <ClInclude Include="..\include\asio\detail\thread_group.hpp" />
+    <ClInclude Include="..\include\asio\detail\thread_info_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\throw_error.hpp" />
+    <ClInclude Include="..\include\asio\detail\throw_exception.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_queue.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_queue_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_queue_ptime.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_queue_set.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_scheduler.hpp" />
+    <ClInclude Include="..\include\asio\detail\timer_scheduler_fwd.hpp" />
+    <ClInclude Include="..\include\asio\detail\tss_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\type_traits.hpp" />
+    <ClInclude Include="..\include\asio\detail\variadic_templates.hpp" />
+    <ClInclude Include="..\include\asio\detail\wait_handler.hpp" />
+    <ClInclude Include="..\include\asio\detail\wait_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winapp_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\wince_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_async_manager.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_async_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_resolver_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_resolve_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_socket_connect_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_socket_recv_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_socket_send_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_ssocket_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_ssocket_service_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_timer_scheduler.hpp" />
+    <ClInclude Include="..\include\asio\detail\winrt_utils.hpp" />
+    <ClInclude Include="..\include\asio\detail\winsock_init.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_event.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_fd_set_adapter.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_fenced_block.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_global.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_handle_read_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_handle_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_handle_write_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_io_context.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_null_buffers_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_operation.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_overlapped_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_overlapped_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_serial_port_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_accept_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_connect_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_recvfrom_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_recvmsg_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_recv_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_send_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_socket_service_base.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_thread_info.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_iocp_wait_op.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_object_handle_service.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_static_mutex.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_thread.hpp" />
+    <ClInclude Include="..\include\asio\detail\win_tss_ptr.hpp" />
+    <ClInclude Include="..\include\asio\detail\work_dispatcher.hpp" />
+    <ClInclude Include="..\include\asio\detail\wrapped_handler.hpp" />
+    <ClInclude Include="..\include\asio\dispatch.hpp" />
+    <ClInclude Include="..\include\asio\error.hpp" />
+    <ClInclude Include="..\include\asio\error_code.hpp" />
+    <ClInclude Include="..\include\asio\execution_context.hpp" />
+    <ClInclude Include="..\include\asio\executor.hpp" />
+    <ClInclude Include="..\include\asio\executor_work_guard.hpp" />
+    <ClInclude Include="..\include\asio\experimental.hpp" />
+    <ClInclude Include="..\include\asio\experimental\co_spawn.hpp" />
+    <ClInclude Include="..\include\asio\experimental\detached.hpp" />
+    <ClInclude Include="..\include\asio\experimental\impl\co_spawn.hpp" />
+    <ClInclude Include="..\include\asio\experimental\impl\detached.hpp" />
+    <ClInclude Include="..\include\asio\experimental\impl\redirect_error.hpp" />
+    <ClInclude Include="..\include\asio\experimental\redirect_error.hpp" />
+    <ClInclude Include="..\include\asio\generic\basic_endpoint.hpp" />
+    <ClInclude Include="..\include\asio\generic\datagram_protocol.hpp" />
+    <ClInclude Include="..\include\asio\generic\detail\endpoint.hpp" />
+    <ClInclude Include="..\include\asio\generic\raw_protocol.hpp" />
+    <ClInclude Include="..\include\asio\generic\seq_packet_protocol.hpp" />
+    <ClInclude Include="..\include\asio\generic\stream_protocol.hpp" />
+    <ClInclude Include="..\include\asio\handler_alloc_hook.hpp" />
+    <ClInclude Include="..\include\asio\handler_continuation_hook.hpp" />
+    <ClInclude Include="..\include\asio\handler_invoke_hook.hpp" />
+    <ClInclude Include="..\include\asio\handler_type.hpp" />
+    <ClInclude Include="..\include\asio\high_resolution_timer.hpp" />
+    <ClInclude Include="..\include\asio\impl\buffered_read_stream.hpp" />
+    <ClInclude Include="..\include\asio\impl\buffered_write_stream.hpp" />
+    <ClInclude Include="..\include\asio\impl\connect.hpp" />
+    <ClInclude Include="..\include\asio\impl\defer.hpp" />
+    <ClInclude Include="..\include\asio\impl\dispatch.hpp" />
+    <ClInclude Include="..\include\asio\impl\execution_context.hpp" />
+    <ClInclude Include="..\include\asio\impl\executor.hpp" />
+    <ClInclude Include="..\include\asio\impl\io_context.hpp" />
+    <ClInclude Include="..\include\asio\impl\post.hpp" />
+    <ClInclude Include="..\include\asio\impl\read.hpp" />
+    <ClInclude Include="..\include\asio\impl\read_at.hpp" />
+    <ClInclude Include="..\include\asio\impl\read_until.hpp" />
+    <ClInclude Include="..\include\asio\impl\serial_port_base.hpp" />
+    <ClInclude Include="..\include\asio\impl\spawn.hpp" />
+    <ClInclude Include="..\include\asio\impl\src.hpp" />
+    <ClInclude Include="..\include\asio\impl\system_context.hpp" />
+    <ClInclude Include="..\include\asio\impl\system_executor.hpp" />
+    <ClInclude Include="..\include\asio\impl\thread_pool.hpp" />
+    <ClInclude Include="..\include\asio\impl\use_future.hpp" />
+    <ClInclude Include="..\include\asio\impl\write.hpp" />
+    <ClInclude Include="..\include\asio\impl\write_at.hpp" />
+    <ClInclude Include="..\include\asio\io_context.hpp" />
+    <ClInclude Include="..\include\asio\io_context_strand.hpp" />
+    <ClInclude Include="..\include\asio\io_service.hpp" />
+    <ClInclude Include="..\include\asio\io_service_strand.hpp" />
+    <ClInclude Include="..\include\asio\ip\address.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v4.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v4_iterator.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v4_range.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v6.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v6_iterator.hpp" />
+    <ClInclude Include="..\include\asio\ip\address_v6_range.hpp" />
+    <ClInclude Include="..\include\asio\ip\bad_address_cast.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_endpoint.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_resolver.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_resolver_entry.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_resolver_iterator.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_resolver_query.hpp" />
+    <ClInclude Include="..\include\asio\ip\basic_resolver_results.hpp" />
+    <ClInclude Include="..\include\asio\ip\detail\endpoint.hpp" />
+    <ClInclude Include="..\include\asio\ip\detail\socket_option.hpp" />
+    <ClInclude Include="..\include\asio\ip\host_name.hpp" />
+    <ClInclude Include="..\include\asio\ip\icmp.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\address.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\address_v4.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\address_v6.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\basic_endpoint.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\network_v4.hpp" />
+    <ClInclude Include="..\include\asio\ip\impl\network_v6.hpp" />
+    <ClInclude Include="..\include\asio\ip\multicast.hpp" />
+    <ClInclude Include="..\include\asio\ip\network_v4.hpp" />
+    <ClInclude Include="..\include\asio\ip\network_v6.hpp" />
+    <ClInclude Include="..\include\asio\ip\resolver_base.hpp" />
+    <ClInclude Include="..\include\asio\ip\resolver_query_base.hpp" />
+    <ClInclude Include="..\include\asio\ip\resolver_service.hpp" />
+    <ClInclude Include="..\include\asio\ip\tcp.hpp" />
+    <ClInclude Include="..\include\asio\ip\udp.hpp" />
+    <ClInclude Include="..\include\asio\ip\unicast.hpp" />
+    <ClInclude Include="..\include\asio\ip\v6_only.hpp" />
+    <ClInclude Include="..\include\asio\is_executor.hpp" />
+    <ClInclude Include="..\include\asio\is_read_buffered.hpp" />
+    <ClInclude Include="..\include\asio\is_write_buffered.hpp" />
+    <ClInclude Include="..\include\asio\local\basic_endpoint.hpp" />
+    <ClInclude Include="..\include\asio\local\connect_pair.hpp" />
+    <ClInclude Include="..\include\asio\local\datagram_protocol.hpp" />
+    <ClInclude Include="..\include\asio\local\detail\endpoint.hpp" />
+    <ClInclude Include="..\include\asio\local\stream_protocol.hpp" />
+    <ClInclude Include="..\include\asio\packaged_task.hpp" />
+    <ClInclude Include="..\include\asio\placeholders.hpp" />
+    <ClInclude Include="..\include\asio\posix\basic_descriptor.hpp" />
+    <ClInclude Include="..\include\asio\posix\basic_stream_descriptor.hpp" />
+    <ClInclude Include="..\include\asio\posix\descriptor.hpp" />
+    <ClInclude Include="..\include\asio\posix\descriptor_base.hpp" />
+    <ClInclude Include="..\include\asio\posix\stream_descriptor.hpp" />
+    <ClInclude Include="..\include\asio\posix\stream_descriptor_service.hpp" />
+    <ClInclude Include="..\include\asio\post.hpp" />
+    <ClInclude Include="..\include\asio\raw_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\read.hpp" />
+    <ClInclude Include="..\include\asio\read_at.hpp" />
+    <ClInclude Include="..\include\asio\read_until.hpp" />
+    <ClInclude Include="..\include\asio\seq_packet_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\serial_port.hpp" />
+    <ClInclude Include="..\include\asio\serial_port_base.hpp" />
+    <ClInclude Include="..\include\asio\serial_port_service.hpp" />
+    <ClInclude Include="..\include\asio\signal_set.hpp" />
+    <ClInclude Include="..\include\asio\signal_set_service.hpp" />
+    <ClInclude Include="..\include\asio\socket_acceptor_service.hpp" />
+    <ClInclude Include="..\include\asio\socket_base.hpp" />
+    <ClInclude Include="..\include\asio\spawn.hpp" />
+    <ClInclude Include="..\include\asio\ssl.hpp" />
+    <ClInclude Include="..\include\asio\ssl\context.hpp" />
+    <ClInclude Include="..\include\asio\ssl\context_base.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\buffered_handshake_op.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\engine.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\handshake_op.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\io.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\openssl_init.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\openssl_types.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\password_callback.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\read_op.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\shutdown_op.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\stream_core.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\verify_callback.hpp" />
+    <ClInclude Include="..\include\asio\ssl\detail\write_op.hpp" />
+    <ClInclude Include="..\include\asio\ssl\error.hpp" />
+    <ClInclude Include="..\include\asio\ssl\impl\context.hpp" />
+    <ClInclude Include="..\include\asio\ssl\impl\src.hpp" />
+    <ClInclude Include="..\include\asio\ssl\rfc2818_verification.hpp" />
+    <ClInclude Include="..\include\asio\ssl\stream.hpp" />
+    <ClInclude Include="..\include\asio\ssl\stream_base.hpp" />
+    <ClInclude Include="..\include\asio\ssl\verify_context.hpp" />
+    <ClInclude Include="..\include\asio\ssl\verify_mode.hpp" />
+    <ClInclude Include="..\include\asio\steady_timer.hpp" />
+    <ClInclude Include="..\include\asio\strand.hpp" />
+    <ClInclude Include="..\include\asio\streambuf.hpp" />
+    <ClInclude Include="..\include\asio\stream_socket_service.hpp" />
+    <ClInclude Include="..\include\asio\system_context.hpp" />
+    <ClInclude Include="..\include\asio\system_error.hpp" />
+    <ClInclude Include="..\include\asio\system_executor.hpp" />
+    <ClInclude Include="..\include\asio\system_timer.hpp" />
+    <ClInclude Include="..\include\asio\thread.hpp" />
+    <ClInclude Include="..\include\asio\thread_pool.hpp" />
+    <ClInclude Include="..\include\asio\time_traits.hpp" />
+    <ClInclude Include="..\include\asio\ts\buffer.hpp" />
+    <ClInclude Include="..\include\asio\ts\executor.hpp" />
+    <ClInclude Include="..\include\asio\ts\internet.hpp" />
+    <ClInclude Include="..\include\asio\ts\io_context.hpp" />
+    <ClInclude Include="..\include\asio\ts\net.hpp" />
+    <ClInclude Include="..\include\asio\ts\netfwd.hpp" />
+    <ClInclude Include="..\include\asio\ts\socket.hpp" />
+    <ClInclude Include="..\include\asio\ts\timer.hpp" />
+    <ClInclude Include="..\include\asio\unyield.hpp" />
+    <ClInclude Include="..\include\asio\uses_executor.hpp" />
+    <ClInclude Include="..\include\asio\use_future.hpp" />
+    <ClInclude Include="..\include\asio\version.hpp" />
+    <ClInclude Include="..\include\asio\waitable_timer_service.hpp" />
+    <ClInclude Include="..\include\asio\wait_traits.hpp" />
+    <ClInclude Include="..\include\asio\windows\basic_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\basic_object_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\basic_random_access_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\basic_stream_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\object_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\object_handle_service.hpp" />
+    <ClInclude Include="..\include\asio\windows\overlapped_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\overlapped_ptr.hpp" />
+    <ClInclude Include="..\include\asio\windows\random_access_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\random_access_handle_service.hpp" />
+    <ClInclude Include="..\include\asio\windows\stream_handle.hpp" />
+    <ClInclude Include="..\include\asio\windows\stream_handle_service.hpp" />
+    <ClInclude Include="..\include\asio\write.hpp" />
+    <ClInclude Include="..\include\asio\write_at.hpp" />
+    <ClInclude Include="..\include\asio\yield.hpp" />
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <ImportGroup Label="ExtensionTargets">
+  </ImportGroup>
+</Project>
\ No newline at end of file
-- 
2.19.0.windows.1

