From 82cf8da4126e9e2d99c0495c5baa4071f25811c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Adrien=20B=C3=A9raud?= <adrien.beraud@savoirfairelinux.com>
Date: Mon, 3 May 2021 16:16:55 -0400
Subject: [PATCH] build: don't force AR path

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 44c076a..8823030 100644
--- a/Makefile
+++ b/Makefile
@@ -124,7 +124,7 @@ LINKED_LIB_SH := lib$(LIB_NAME).$(LINKED_LIB_EXT)
 endif
 
 # Some systems don't provide an unprefixed ar when cross-compiling.
-AR=ar
+AR ?= ar
 
 LIBRARIES = $(LIB_SH) $(LIB_ST)
 HEADERS = include/argon2.h
-- 
2.27.0

